analytics_data_api/v0/models.py

```diff
@@ -242,8 +242,22 @@ def get_users_in_course(
         the Search object.  Raises `ValueError` if both `segments` and
         `ignore_segments` are provided.
         """
+        # Error handling
         if segments and ignore_segments:
             raise ValueError('Cannot combine `segments` and `ignore_segments` parameters.')
+        for segment in (segments or list()) + (ignore_segments or list()):
+            if segment not in learner.SEGMENTS:
+                raise ValueError(
+                    'segments/ignore_segments values must belong to: {}'.format(', '.join(learner.SEGMENTS))
```

Comment:
It might be helpful to state the invalid segment.  Maybe something like: `segments/ignore_segments value ('{segment}') must belong to: {the_list}`